plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/decoders/DecoderFactory.java

```diff
@@ -50,7 +51,7 @@ public static Decoder createDecoder(Type type)
             if (type instanceof BigintType) {
                 return new BigintDecoder();
             }
-            if (type instanceof IntegerType) {
+            if (type instanceof IntegerType || type instanceof DateType) {
```

Comment:
Pinot doesn't support dates > Integer.MAX_VALUE epoch?

Reply:
It does, although that is not common in practice. If a date value is > `Integer.MAX_VALUE`, then the conversion to trino `DateType` will fail. There is a todo in `DateType` to change to long.